### PR TITLE
feat(python): support python requirements.txt

### PR DESCRIPTION
**Pull request by Sam-Lane**
Raising to help support this ticket in Trivy.

https://github.com/aquasecurity/trivy/issues/492

Thank you.

### CODE REVIEW

**Review by hubseven — APPROVED** [review]

File: pkg/python/testdata/requirements_flask.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+click==8.0.0
+Flask==2.0.0
+itsdangerous==2.0.0
```
Comment (masahiro331) [inline pkg/python/testdata/requirements_flask.txt:3]: https://pip.pypa.io/en/latest/cli/pip_install/#requirements-file-format
The following patterns should also be supported.
```
itsdangerous==2.0.0 # a comment 
# itsdangerous==2.0.0
```

File: pkg/python/parse.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		if len(s) != 2 {
+			continue
+		}
+		libs = append(libs, types.Library{
+			Name:    s[0],
+			Version: s[1],
+		})
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("scan error: %w", err)
```
Comment (knqyf263) [inline pkg/python/parse.go:26]: For stacktrace

```suggestion
		return nil, xerrors.Errorf("scan error: %w", err)
```

File: pkg/python/parse_test.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+				return ret < 0
+			})
+
+			sort.Slice(v.want, func(i, j int) bool {
+				ret := strings.Compare(v.want[i].Name, v.want[j].Name)
+				if ret == 0 {
+					return v.want[i].Version < v.want[j].Version
+				}
+				return ret < 0
+			})
```
Comment (knqyf263) [inline pkg/python/parse_test.go:49]: Looks unnecessary.

File: pkg/python/parse_test.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+func TestParse(t *testing.T) {
+	vectors := []struct {
+		file string // Test input file
+		want []types.Library
+	}{
+		{
+			file: "testdata/requirements_flask.txt",
+			want: requirementsFlask,
```
Comment (knqyf263) [inline pkg/python/parse_test.go:23]: Please add more tests. `requirements.txt` may include requirements without version specifiers, other requirements files, etc.
https://pip.pypa.io/en/latest/cli/pip_install/#requirements-file-format

Comment (knqyf263) [reply pkg/python/parse_test.go:23]: Also, the version may have spaces after `==` as follows.
```
docopt == 0.6.1
```

File: pkg/python/testdata/requirements_flask.txt (hunk 1 above)
Comment (Sam-Lane) [reply pkg/python/testdata/requirements_flask.txt:3]: Added comment stripping and test coverage for this. Let me know if further test cases are required.

File: pkg/python/parse_test.go (hunk 4 above)
Comment (Sam-Lane) [reply pkg/python/parse_test.go:21]: Added whitespace stripping and test coverage for this. Let me know if more is required.

**Comment by masahiro331** [comment]
@Sam-Lane 
Sorry late response, Looks good. 

### Feature
Could you support this specification? 
https://pip.pypa.io/en/latest/cli/pip_install/#requirement-specifiers
https://www.python.org/dev/peps/pep-0508/#id16

No support for comparison operators other than "==" is required.

### Test
And add test more format. e.g
```
keyring >= 4.1.1            # Minimum version 4.1.1
coverage != 3.5             # Version Exclusion. Anything except version 3.5
Mopidy-Dirble ~= 1.1        # Compatible release. Same as >= 1.1, == 1.*
```

File: pkg/python/parse.go [hunk 5]
```diff
@@ -34,8 +36,8 @@ func Parse(r io.Reader) ([]types.Library, error) {
 	return libs, nil
 }
 
-func stripComments(line *string) {
-	if pos := strings.IndexAny(*line, commentRune); pos >= 0 {
+func RstripByKey(line *string, key string) {
```
Comment (knqyf263) [inline pkg/python/parse.go:39]: Why do you need to export this function? And I prefer returning updated text like `strings.TrimPrefix`.

Comment (ankk13) [reply pkg/python/parse.go]: Sorry, it was a mistake

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**